### PR TITLE
C_CPP : Forward List ( STL ) in document form

### PR DESCRIPTION
**Pull request by Pranshu321**
<hr>

## Description 📜

In this, I deeply explain that 
1. What is the forward list?
2. How does it differ from the list?
3. Its pros, cons, and application.
4. Functions used in forward list.

Issue number closed : #6921 

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Review by gunjan-g — APPROVED** [review]
Great work @Pranshu321 
LGTM 🚀  @Saviour1001 @Suvraneel

File: C_CPP/Data Structures/Standard Template Library/Forward_List.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+1. assign : assign function is used when we have to assign the value to list , we can also assign repeated value by this function .
+```cpp
+// Assigning values to the list using assign() function
+	list_1.assign({5, 4, 7});
+
+// Pushing 3 elements with value 40
+	list_2.assign(3,4);
+```
+
+2. push_front : It is used to push the element in the starting of the list .
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Standard Template Library/Forward_List.md:54]:
```suggestion
2. **push_front** : It is used to push the element in the starting of the list .
```

Bolden the keywords... Makes it more legible

Comment (Suvraneel) [reply C_CPP/Data Structures/Standard Template Library/Forward_List.md:54]: All of them...

**Review by Suvraneel — CHANGES_REQUESTED** [review]
Embed an image depicting the structure as well

File: C_CPP/Data Structures/Standard Template Library/Forward_List.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
+# Forward List
+Forward Lists is same as the list in STL ( Standard Template Library ) , But it differ in one aspect that in this element only keep the track of next element not of previous but in list the element keep track of both next and previous element .
+## Advantages Over List
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Standard Template Library/Forward_List.md:3]: Add a line before 2nd level head

**Review by Suvraneel — CHANGES_REQUESTED** [review]

**Comment by Suvraneel** [comment]
@Pranshu321  LGTM ❤️ 🚀  :octocat:

**Pull request merged**